magento/graphql-ce
# Respect allow send as guest configuration in resolver

### Description (*)
Resolver throws `GraphQlAuthorizationException` when sending as guest if allow
send as guest is disabled.

### Fixed Issues (if relevant)
1. magento/graphql-ce#732

### Manual testing scenarios (*)
Send a Friend Mutation:
```graphql
mutation {
  sendEmailToFriend(
    input: {
      product_id: {$productId}
      sender: {
        name: "Name"
        email: "e@mail.com"
        message: "Lorem Ipsum"
      }          
      recipients: [{
        name: "Recipient Name 1"
        email:"recipient1@mail.com"
      }]
    }
  ) {
    sender {
      name
      email
      message
    }
    recipients {
      name
      email
    }
  }
}
```
1. Vanilla Magento install. Default value for Stores > Configuration > Catalog > Email a Friend > Allow For Guests is No
2. Execute above mutation as guest.
3. Validate `GraphQlAuthorizationException`
4. Authenticate as customer
5. Execute mutation including authentication token
6. Validate success response
7. Set Allow for Guests to Yes
8. Repeat items 2-6. Guest mutation should no longer return `GraphQlAuthorizationException`

### Contribution checklist (*)
